### ISSUE TITLE
feat: default allowed_gas

### ISSUE THREAD

**Issue by milancermak** [starkware-libs/cairo#4017]
# Feature Request

Add an attribute that sets the `available_gas` for every `#[test]` in a module.

**Describe Preferred Solution**

Introduce a new module attribute `default_available_gas` that would set the available gas for each test case in the decorated module. If a test case has its specific `available_gas` attribute, the test runner would use this specific value to set the gas.

```rust
#[cfg(test)]
#[default_available_gas(1000000000)]
mod tests {
    #[test]
    fn test_1() {
        // uses the mod default available gas of 1000000000
    }

    #[test]
    #[available_gas(1000)]
    fn test_perf() {
        // uses the test specific available gas of 1000
    }
}
```

**Describe Alternatives**

An alternative would be for the test runner to accept a `--available-gas` cmdline argument that would act in the same way as the attribute described above. It could also work in tandem where the cmdline arg would be a less specific value. The available gas resolution would then be from most to least specific - test case attribute, module attribute, command line arg, global default.

**Related Code**

**Additional Context**

Pretty much everyone just amends their test cases with `available_gas` and a random big number to have their test cases running. Examples from [Kakarot](https://github.com/kkrt-labs/kakarot-ssj/blob/main/crates/evm/src/tests/test_instructions/test_block_information.cairo), [Renegade finance](https://github.com/renegade-fi/renegade-contracts/blob/main/src/testing/tests/merkle_tests.cairo) or [this repo](https://github.com/starkware-libs/cairo/blob/main/corelib/src/test/byte_array_test.cairo). It's annoying and having a way how to globally specify the number would result in a much better devex.

**If the feature request is approved, would you be willing to submit a PR?**
_(Help can be provided if you need assistance submitting a PR)_

- [x] Yes

I'd need some guidance.

**Comment by milancermak** [comment]
> Should user be unaware for their tests that they take gas?

IMO the most common use case is "I just want to ensure my tests run, I want to test my code." 95% of the time, these tests are not performance tests, but unit or integration tests. So I'd say it's fine if users (developers) are not aware that tests take gas. In which case, your proposed solution of infinity available gas is great.

For the remaining cases where we want to test the efficiency of a function, the current `available_gas` attribute works fine.

**Comment by orizi** [comment]
so we also probably need to add some option for "const_gas" - to enable ensuring a function does not take gas.
suggestions for such a name?

**Comment by milancermak** [comment]
Honestly, I suck at naming things 😬

**Comment by milancermak** [comment]
Resolved by #4172 ♥️
